packages/integration-tests/projects/connect-popup/index.js

```diff
@@ -0,0 +1,370 @@
+    //             ],
+    //         },
+
+
+    //     ]
+    // }
+]
+
+
+const wait = (timeout) => {
+    return new Promise((resolve) => {
+        setTimeout(() => {
+            resolve()
+        }, timeout)
+    })
+}
+
+const log = (...val) => {
+    console.log(`[===]`, ...val);
+}
+
+(async () => {
+    try {
+        if (fs.existsSync(SCREENSHOTS_DIR)) {
+            fs.rmdirSync(SCREENSHOTS_DIR, { recursive: true });
+            fs.mkdirSync(SCREENSHOTS_DIR, { recursive: true });
+        } else {
+            fs.mkdirSync(SCREENSHOTS_DIR, { recursive: true });
+        }
+
+        const controller = new Controller({ url: 'ws://localhost:9001/' });
+
+        let retries = 0;
+        let connected = false;
+        while (!connected && retries < 60) {
+            try {
+                await controller.connect();
+                connected = true;
+            } catch (err) {
+                console.log('waiting for trezor-user-env...');
+            }
+            await wait(1000);
+            retries++;
+        }
+
+        const browser = await puppeteer.launch({
+            headless: HEADLESS,
+            args: [
+                "--disable-gpu",
+                "--disable-dev-shm-usage",
+                "--disable-setuid-sandbox",
+                "--no-sandbox",
+            ]
+        });
+
+        // todo: another case of buggy behavior in bridge 3.0.31.
+        // await controller.send({ type: 'bridge-start', version: '2.0.27' });
+
+        for (setup of setups) {
+            await wait(500)
+            await controller.send({
+                type: 'emulator-stop',
+            });
+            await wait(500)
+            await controller.send({
+                type: 'emulator-start',
+                wipe: true,
+            });
+            await wait(500)
+            await controller.send({
+                type: 'emulator-setup',
+                ...setup.device,
+            });
+            await wait(500)
+            await controller.send({ type: 'emulator-allow-unsafe-paths' });
+            await wait(500)
+
+            for (f of setup.fixtures) {
+
+                await controller.send({ type: 'bridge-stop' });
+                await wait(1000)
+                await controller.send({ type: 'bridge-start', version: '2.0.27' });
+                await wait(1000)
+
+
+                log(f.method, "start");
+
+                const screenshotsPath = `${SCREENSHOTS_DIR}/${f.method}`;
+
+                if (!fs.existsSync(screenshotsPath)) {
+                    fs.mkdirSync(screenshotsPath);
+                }
+
+                const explorer = await browser.newPage();
+                const popupPromise = new Promise(x => explorer.once('popup', x));
+
+                await explorer.goto(`${url}#/method/${f.method}`);
+
+                // screenshot request
+                log(f.method, "screenshot trezor-connect call params");
+                await explorer.waitForTimeout(2000);
```

Comment:
What is the reason you are using `wait` and `waitForTimeout` in different places?

Reply:
`wait`s are gone. still couple of instances of `waitForTimeout` which I would like to remove